cip/CIP2015-10-12-CREATE.adoc

```diff
@@ -0,0 +1,136 @@
+Please note that `CREATE UNIQUE` and `CREATE CONSTRAINT` are _not_ considered in this CIP.
+--
+
+toc::[]
+
+== Motivation & Background
+`CREATE` is how nodes and relationships are created in Cypher.
+The syntax has been around since Cypher 1.7 but never properly formalized.
+
+== Proposal
+
+`CREATE` is used to add new nodes and relationships to the graph, as well as set initial labels, type, and properties on the new entities.
+The idea is to let each `CREATE` represent exactly one creation of either a node or a relationship, but allow a few variations on how to structure the query for convenience.
+`CREATE` will be allowed to both start and end a query, requiring an additional `WITH` clause to allow further `MATCH` reads after it, and producing an empty result if it ends the query.
+`MATCH` clauses that come after a `CREATE` clause must be able to see and possibly match _all_ entities created by the `CREATE`; not only those created for the same row.
+
+A `CREATE` will always succeed unless it specifically violates some constraint already present in the database.
+
+=== Examples
+
+[source, cypher]
+.Create a single node.
+----
+CREATE ()
+----
+
+[source, cypher]
+.Create a node and set two labels and one property on it.
+----
+CREATE (:Person:Administrator { email: 'mats@neotechnology.com' })
+----
+
+[source, cypher]
+.Create a relationship of type `KNOWS` between all `:Person` nodes.
+----
+MATCH (a:Person), (b:Person)
+CREATE (a)-[:KNOWS]->(b)
+----
+
+[source, cypher]
+.Create two nodes, one relationship between them, and sets one property on the relationship.
+----
+CREATE (me:Source), (you:Destination), (you)<-[:LOVES { dearly: true }]-(me)
+----
+
+[source, cypher]
+.Create multiple nodes with properties from a parameter list of maps.
+----
+UNWIND { mapList } AS map
+CREATE (n:Node)
+SET n = map
+----
+
+[source, cypher]
+.Create two nodes with properties from two parameter maps. Parameter keys and values will be the same as those of the map.
+----
+CREATE (:Cat {catProps})
+CREATE (:Dog {dogProps})
+----
+
+[source, cypher]
+.Create three nodes, connect them with two relationships and return the whole pattern as a path.
+----
+CREATE (andres { name: 'Andres' })
+CREATE (neo)
+CREATE (michael { name: 'Michael' })
+CREATE (andres)-[r1:WORKS_AT]->(neo)
+CREATE (michael)-[r2:WORKS_AT]->(neo)
+RETURN (andres)-[r1]->(neo)<-[r2]-(michael) AS p
+----
+
+=== Syntax
+[source, ebnf]
+----
+create       = "CREATE", {pattern}, [{",", pattern}] ;
+pattern      = node-pattern | rel-pattern ;
+node-pattern = "(", [identifier], [{label}], [property], ")" ;
+rel-pattern  = "(", identifier, ")", in-rel | out-rel, "(", identifier, ")" ;
+out-rel      = "-[", [identifier], type, [property], "]->" ;
+in-rel       = "<-[", [identifier], type, [property], "]-" ;
+label        = ":", identifier ;
+type         = label ;
+property     = "{", param | {identifier, ":", literal | param}, "}" ;
```

Comment:
the `param` production already contains the `"{"` and `"}"`, and should this be on the "top level" of the `property` production.